changes/2020-06-17_encrypt-max-plaintext-length-input/change.md

```diff
@@ -0,0 +1,203 @@
+
+The specification already specifies a [plaintext length](../../client-apis/encrypt.md#plaintext-length)
+input on the Encrypt operation that describes a bound on the plaintext length,
+however its effect on the Encrypt operation's behavior is unspecified.
+
+This change renames this input to `plaintext length bound` in the specification
+to be a more accurate reflection of its behavior and intent,
+and specifies the correct behavior during the Encrypt operation.
+Specifically, in the scope of the Encrypt operation it's value MUST be passed to the CMM.
+
+Additionally, it is unclear in the spec what the behavior should be
+if the input plaintext has a known-length
+and the user specifies a bound on the length of that plaintext.
+
+This change proposes that in the scope of the Encrypt operation,
+an input plaintext is either of known length,
+of unknown length with user supplied estimated size,
+or unknown length with no user supplied estimated size.
+The Encrypt operation should never consider a `plaintext length bound` alongside
+a known-length plaintext.
+
+How a user should express on the API level the intent to encrypt a known-length plaintext
+or unknown-length plaintext with or without a bound on the length
+depends on the specific implementation
+and the idioms of that implementation's language to describe APIs.
+
+## Out of Scope
+
+- Significantly changing the shape of any of our Encrypt API implementations is out of scope.
+
+## Motivation
+
+All implementations allow users to specify a plaintext length when calling Encrypt with a
+unknown-length plaintext.
+However, its behavior in the Encrypt operation is not clearly specified.
+
+Additionally, some implementations also allow a plaintext length to be specified on Encrypt with a
+known-length plaintext.
+The behavior of Encrypt in this case is not consistent between implementations.
+
+The purpose of this change is to specify exactly how this input affects the Encrypt operation.
+
+This change renames this input to `plaintext length bound` within the spec because
+the intention behind the value of this input is that
+it is a max bound of the plaintext length,
+set by the user when streaming encryption.
+If the actual plaintext length is greater than this value,
+something is wrong and the operation MUST fail.
+Otherwise, the Encrypt operation MUST pass this value to the CMM.
+
+The intention of this value is not to be a one to one passthrough value to the CMM.
+If that were the case,
+then we would be concerned with always letting users specify this value for known-length plaintexts,
+or with possibly letting users pass in a "unknown length" intent.
+We do not want to support these cases.
+Users should not be setting a `max plaintext length` value for known-length plaintexts
+with the intention of changing CMM behavior.
+Instead, users should be specifying this value in for unknown-length plaintexts in order to use CMMs
```

Comment:
Users `should`/`should not` or users `must`/`must not`?

Reply:
The first is definitely a SHOULD NOT, which makes the latter be a SHOULD. Alternatively, I'd be ok with using a different verb to avoid RFC 2119 verbs when not strictly necessary.